capsules/src/lsm303agr.rs

```diff
@@ -374,7 +429,7 @@ impl i2c::I2CClient for Lsm303agrI2C<'_> {
                 self.i2c_accelerometer.disable();
                 self.state.set(State::Idle);
                 if self.config_in_progress.get() {
-                    self.set_magneto_data_rate(self.mag_data_rate.get());
+                    let _ = self.set_magneto_data_rate(self.mag_data_rate.get());
```

Comment:
Could this cause things to hang if the I2C is busy for some reason? I'm a little wary of the potential error being discarded. If you have reason to think that this should not happen, then please add a comment explaining why.

Reply:
Thank you for pointing this out.